lib/gollum/views/commit.rb

```diff
@@ -23,12 +23,24 @@ def message
       end
 
       def files
-        files = @diff.split(%r{^diff --git a/.+ b/.+$}).reject(&:empty?)
+        files = @diff.force_encoding(Encoding::UTF_8).scan(%r{
```

Comment:
Nice and clean refactor! Only thing I'm wondering about is: does this still implement the equivalent of `.reject(&:empty?)`? Or do we no longer need it?

Reply:
Also, are the changes in this file just a refactor, or do they implement (slightly) different behavior?

Reply:
Ah, I think I see it now: we're precisely not rejecting empty diffs to support changes to binary files?

Reply:
> does this still implement the equivalent of .reject(&:empty?)
I don't think we need it anymore because we use a scan method, so there won't be empty strings.

Reply:
> Also, are the changes in this file just a refactor, or do they implement (slightly) different behavior?

Mainly it's a refactor, but I also [added another `gg` class](
https://github.com/gollum/gollum/pull/1649/commits/c93a46034b220e99a5ff1c134f883e29436fe583) for `\ No newline...` and `Binary files...` messages and [put `...` as placeholder for line numbers.](https://github.com/gollum/gollum/pull/1649/files#diff-09095607da570d2efe65488d68a6b0d6d3cfd6c5661d489855835572abd4946fL78-R85)